File: Solution Sorting.js

/**
 * @param {string} inputString
 * @return {number}
 */
var minDeletions = function (inputString) {

    const ALPHABET_SIZE = 26;
    const ASCII_SMALL_CASE_A = 97;
    const frequency = new Array(ALPHABET_SIZE).fill(0);
    for (let i = 0; i < inputString.length; ++i) {
        ++frequency[inputString.codePointAt(i) - ASCII_SMALL_CASE_A];
    }

    frequency.sort((x, y) => x - y);
    let maxFrequency = inputString.length;
    let countDeletions = 0;

    for (let i = ALPHABET_SIZE - 1; i >= 0; --i) {
        if (frequency[i] > maxFrequency) {
            countDeletions += frequency[i] - maxFrequency;
            frequency[i] = maxFrequency;
        }
        maxFrequency = Math.max(0, frequency[i] - 1);
    }
    return countDeletions;
};

File: Solution Sorting.cpp

#include <string>
#include <vector>
using namespace std;

class Solution {
    
    inline static const int ALPHABET_SIZE = 26;

public:
    int minDeletions(const string& inputString) {

        array<int, ALPHABET_SIZE> frequency{};
        for (int i = 0; i < inputString.length(); ++i) {
            ++frequency[inputString[i] - 'a'];
        }

        sort(frequency.begin(), frequency.end());
        int maxFrequency = inputString.length();
        int countDeletions = 0;

        for (int i = ALPHABET_SIZE - 1; i >= 0; --i) {
            if (frequency[i] > maxFrequency) {
                countDeletions += frequency[i] - maxFrequency;
                frequency[i] = maxFrequency;
            }
            maxFrequency = max(0, frequency[i] - 1);
        }
        return countDeletions;
    }
};
